metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -215,8 +215,20 @@ func getNetworkNames(c *vim25.Client, ref types.ManagedObjectReference) ([]strin
 		return nil, errors.Wrap(err, "no networks found")
 	}
 
+	networksRefs := []types.ManagedObjectReference{}
+	for _, obj := range vm.Network {
+		if obj.Type == "Network" {
+			networksRefs = append(networksRefs, obj)
+		}
+	}
+
+	// If only "Distributed port group" was found, for example.
+	if len(networksRefs) == 0 {
+		return nil, errors.Wrap(err, "no networks found")
```

Comment:
I think you want `errors.New` instead of wrap because err == nil here. Same thing with the earlier errors.wrap usage.

Reply:
Changed!